sources/packages/web/src/components/partial-view/DesignationAgreement/DesignationAgreementForm.models.ts

```diff
@@ -46,3 +46,29 @@ export interface DesignationModel {
   designationStatus?: DesignationAgreementStatus;
   designationStatusClass?: string;
 }
+
+/**
+ * Details for approve or update an approved designation agreement.
+ */
+export interface UpdateDesignationDetailsModel {
+  designationStatus: DesignationAgreementStatus;
+  startDate?: string;
```

Comment:
Just  to double-check, won't startDate and endDate are required during the approval process?

Reply:
Yes, they are. This model is used to map the UI, from this model the DTO will be created and there they should be required.